mofiebiger/ember-trivia
# New <QuestionCard /> component and Ember Data

# ✨ What Changed & Why

There are _5_ main changes handled in this PR. Again... this is WAAAAY to many aspects for production!

- **Updated style/behaviour to `<UiHeader />` and `<StartGamePage />`:**
	- Here `<UiHeader />` now returns a `p` tag instead of and `h1` with a size of `25px`. This way header text size can be overridden if needed. 
	- Next a _pointer_ was added to the `<StartGamePage />` drop downs and the `Play` button route is update to `questions.question-one`.
- **New Global Adapter Logic:**
	A global adapter page is added here (`application.js`). Firstly defining the `namespace=api`. This means in the all data will be the file system under _./public/api_ and _http://localhost:4200/api_ in the browser. So essentially we are just telling Ember where to look for our data. Second, in the build we append .json to the end of our file name. So in the browser it would look like this: http://localhost:4200/api/questions-data.json. For more info on this please visit this page: https://guides.emberjs.com/release/tutorial/part-2/ember-data/#toc_working-with-adapters-and-serializers.
- **Ember Data:**
	- JSON: A handmade json file specifically tailored for ember data. This is a work in progress and a collection of ember trivia questions found from the internet/personal knowledge. 
	- Model: This file declares all the `@attrs` used 
		- @attr category;
  		- @attr difficulty;
  		- @attr question;
		- @attr correct_answer;
		- @attr options;
		- @attr question_type;
		- @attr score;
- **New _Questions_ route with siblings:**
	- One main route: `questions` with three children: `Q1`, `Q2`, `Q3`. `Questions` takes in the _question model_ and filters the data by _category_ and _difficulty_. From here, the subsequent children display the one object (data[0] - data[2]). Then this data is passed into the `<QuestionCard />` and rendered on the screen.
- **New `<QuestionCard />` component:**
This component is used on all three questions and takes in the model input to work on its logic. To start this component displays the question and all four options. Additional logic is used to style the button pressed; if the selected button is correct it displays green and if the selected button is incorrect it displays red. 


# 👹 Before

![startgamepage](https://user-images.githubusercontent.com/43208685/86483765-a57fc600-bd4c-11ea-9d25-b2e451c24fe1.gif)

# 😇 After

![afterFlowEMBERT](https://user-images.githubusercontent.com/43208685/88398759-c1134500-cdbd-11ea-88ff-16bed6d30a7f.gif)

# 🌈 Testing

1. Visit [Ember-Trivia-Website](https://deploy-preview-8--kind-haibt-4424bd.netlify.app/)
1. Select _Category_ and _Difficulty_ from downdown
1. Select `Play`
1. Notice new question cards
1. Observe api data passed to question cards
